Add documentation note about rootCloseEvent being unavailable to custom dropdowns

kevinzwhuang opened the pull request:
Summary
-----------
The documentation about `rootCloseEvent` for Dropdown does not indicate that it is not usable for `CustomDropdown`.

Changes proposed
-------------------
- [x] Add note about `rootCloseEvent` not working for custom dropdowns due it being implemented internally by `<Dropdown.Menu>`.

-------------------
Fixes #2891

taion commented:
Could you just say that users need to explicitly pass this down to the `<RootCloseWrapper>`?

kevinzwhuang commented:
@taion So telling users to import `<RootCloseWrapper>` from react-overlay themselves in their custom dropdowns and passing down the event there?

taion commented:
I think so! Is it in the formal docs section on custom menus?

kevinzwhuang commented:
Doesn't look like it - there hasn't been a public mention of `<RootCloseWrapper>` in the docs thus far (did a global search in the docs and the codebase, only appears in the `src/DropdownMenu.js` source, not the docs)

We could also add a mention there in the `## Custom Dropdown Components` section recommending the usage of `<RootCloseWrapper>` for this

kevinzwhuang commented:
@taion This seems like it might be a little difficult for most users to reimplement directly using `<RootCloseWrapper>`...but here's a try at some notes about it here:

----------------

Note: To simulate the `rootCloseEvent` prop for custom dropdown menu components, you will need to import `<RootCloseWrapper>` from [react-overlays](https://react-bootstrap.github.io/react-overlays/) and render it as the top node of your custom menu component. Then pass the `rootCloseEvent` to `<RootCloseWrapper>` as the event prop it expects ([similar to how `<Dropdown.Menu>` is implemented](https://github.com/react-bootstrap/react-bootstrap/blob/e41e041709045db657ac02c46fa4ff2df4e339a3/src/DropdownMenu.js#L115-L119)).

How it looks in that section: (I also started writing out some example code, but I feel like that might make it more complicated, maybe they should just look at DropdownMenu and reimplement it)
![image](https://user-images.githubusercontent.com/7256178/33917562-e7e97276-df63-11e7-8e3a-eae9eeb58ae7.png)

taion commented:
It might be good to show this in the actual example, since we already have that. Though maybe the import from react-overlays won't work.

Maybe just mention it briefly for now, and we can clean it up once we merge the Gatsby docs.

kevinzwhuang commented:
For sure @taion - I've added callout to this in the custom dropdown component section and updated the note in the props section to hint at this.

Custom Dropdown Components section:
![image](https://user-images.githubusercontent.com/7256178/34057281-286fdd98-e18b-11e7-9ed6-4059dcdd46ec.png)

Dropdown Props:
![image](https://user-images.githubusercontent.com/7256178/34057289-314e4a44-e18b-11e7-8505-0845c94b810c.png)

taion commented on docs/src/sections/DropdownSection.js:
```diff
@@ -57,6 +57,15 @@ export default function DropdownSection() {
+      <div className="bs-callout bs-callout-warning">
+        <h4>Using <code>rootCloseEvent</code> with custom dropdown components</h4>
+        <p>
+          If you want to use the <code>rootCloseEvent</code> prop with your custom dropdown components, you will have to pass the <code>rootCloseEvent</code> to <code>{ '<RootCloseWrapper>' }</code> in your custom dropdown menu component <a href="https://github.com/react-bootstrap/react-bootstrap/blob/e41e041709045db657ac02c46fa4ff2df4e339a3/src/DropdownMenu.js#L115-L119">similarly to how it is i ...
```
Use a tag instead of the commit here, like https://github.com/react-bootstrap/react-bootstrap/blob/v0.31.5/src/DropdownMenu.js#L115-L119

taion commented on src/Dropdown.js:
```diff
@@ -95,6 +95,10 @@ const propTypes = {
+   *
+   * *Note: For custom dropdown components, you will have to pass the
+   * `rootCloseEvent` to `<RootCloseWrapper>` in your custom dropdown menu
+   * component ([similarly to how it is implemented in `<Dropdown.Menu>`](https://github.com/react-bootstrap/react-bootstrap/blob/e41e041709045db657ac02c46fa4ff2df4e339a3/src/DropdownMenu.js#L115-L119)).*
```
and here

kevinzwhuang commented:
Good call @taion, merging after it passes TravisCI
